internal/rbd/replication.go

```diff
@@ -126,3 +98,23 @@ func (rv *rbdVolume) DisableVolumeReplication(
 
 	return nil
 }
+
+func (rv *rbdVolume) CheckImageIsPrimary() error {
```

Comment:
I am confused by the naming of the function. I expected a true/false return in case the image is (not) primary...

Instead, if the image is primary, it returns a `ErrInvalidArgument`?

What is the intention of this function, and can it be made clearer by renaming and/or returning a bool? A comment with a description of the function would also help.

Reply:
things got messed up when I tried to refractor, for now removed refractoring as it looks complicated to do as part of this PR.